src/SQL/Builder/WithSQLBuilder.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\SQL\Builder;
+
+use Doctrine\DBAL\Query\With;
+
+use function array_map;
+use function count;
+use function implode;
+use function sprintf;
+
+final class WithSQLBuilder
+{
+    public function buildSQL(With ...$withParts): string
```

Comment:
This method can be invoked with an empty set of arguments and will produce knowingly invalid SQL. We could enforce the contract statically by declaring the signature as follows:
```suggestion
    public function buildSQL(With $firstExpression, With ...$otherExpressions): string
```